### PR TITLE
[Reco] Removed unnecessary include statements

### PR DESCRIPTION
**Pull request by smuzaffar**
This should fix the following private header usage issue #34718 for
```
   1 src/RecoLocalCalo/HGCalRecProducers/plugins/KernelManagerHGCalCellPositions.h
```

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: RecoLocalCalo/HGCalRecProducers/plugins/HeterogeneousHGCalHEFCellPositionsConditions.cc [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include "RecoLocalCalo/HGCalRecProducers/plugins/KernelManagerHGCalCellPositions.h"
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecProducers/plugins/HeterogeneousHGCalHEFCellPositionsConditions.cc:1]:
```suggestion
#include "KernelManagerHGCalCellPositions.h"
```
to rely more clearly on locality of the file

**Comment by smuzaffar** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by tvami** [comment]
+alca

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34866	f87a446
- code changes are technical, in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by slava77** [comment]
+1

my previous was ignored

**Comment by perrotta** [comment]
+1
- @cms-sw/upgrade-l2 this is technical, and quite simple: let try to merge it

**Comment by perrotta** [comment]
merge

**Pull request merged**